src/test/regress/sql/shard_rebalancer.sql

```diff
@@ -1359,6 +1392,18 @@ FROM pg_dist_shard
 JOIN pg_dist_shard_placement USING (shardid)
 WHERE logicalrelid = 'r1'::regclass;
 
+SELECT replicate_reference_tables();
+
+-- After replication, the move should succeed.
+SELECT citus_move_shard_placement(pg_dist_shard.shardid, nodename, nodeport, 'localhost', :worker_2_port)
+    FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
+    WHERE logicalrelid = 'd1'::regclass AND nodename = 'localhost' AND nodeport = :worker_1_port LIMIT 1;
+
+-- And a JOIN with the reference table should work.
+SELECT 1 as dummy FROM d1 JOIN r1 USING (a);
```

Comment:
This should be moved up to, right above `DROP TABLE d1, r1;`

Reply:
Why do you think so? This is the query that fails when you move a shard but don't replicate reference tables. I intended it to run here to prove that the JOIN can be performed after `citus_move_shard_placement`. It may be pointless when we explicitly do `replicate_reference_tables`, though. So maybe I could just remove it. What do you think?

Reply:
to be clear, I meant to that you should move up this whole block.

Reply:
Sorry, that means I have messed up. I'll clean it up but it will take at least a week before I can get back to it and focus properly - just FYI that I'm not giving up, just out of time due to other priorities.

Reply:
You were right. I tried to clean it up. Please have a look at the current state.